src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -486,13 +486,15 @@ public String getContentDisplay()
                     name = user.getName();
                 tmp = tmp.replaceAll("<@!?" + Pattern.quote(user.getId()) + '>', '@' + Matcher.quoteReplacement(name));
             }
-            for (Emote emote : getEmotes())
+            for (TextChannel mentionedChannel : getMentionedChannels())
             {
-                tmp = tmp.replace(emote.getAsMention(), ":" + emote.getName() + ":");
+                tmp = tmp.replaceAll(
+                        "<#" + mentionedChannel.getId() + "(?::(\\d+):([^>\\s]+))?>", // possible mention formats
```

Comment:
Do you think it would be possible to somehow reuse the regex we already have instead of having it duplicated twice as a literal?

Reply:
Would be possible but probably not worth the effort. Its unlikely this will ever be changed again.